packages/components/src/button/index.tsx

```diff
@@ -49,6 +49,10 @@ function useDeprecatedProps( {
 	};
 
 	if ( isSmall ) {
+		deprecated( 'Button isSmall prop', {
```

Comment:
> Perhaps the deprecated warning itself is unnecessary?

Yes, I [personally don't think](https://github.com/WordPress/gutenberg/pull/59913#discussion_r1526612358) it's worth it in this particular case, but I'm not going to block it if you or others think it's warranted. I will suggest adding a Dev Note for the next WP release though.

_If_ we're going to throw a warning, we should add a prefix:

```suggestion
		deprecated( 'wp.components.Button isSmall prop', {
```

Reply:
I decided to remove the `deprecated()` function for now so it doesn't throw any warnings. In the future, we may need to consider adding warnings along with other props such as `isPrimary|isTertiary|isSecondary|isLink`.